bodybuilder.d.ts

```diff
@@ -335,11 +335,36 @@ declare namespace bodybuilder {
 		getRawAggregations(): object;
 	}
 
+	/**
+	 * Options to build a suggestion.
+	 *
+	 * @interface SuggestOptions
+	 * @field name A custom name for the suggestion, defaults to suggest_<type>_<field>.
+	 * @field name Name of predefined analyzer to use on suggest
+	 * @field text Text to suggest on
+	 */
+	export interface SuggestOptions {
+		text: string;
+		analyzer?: string;
+		name?: string;
+	}
+
+	export interface PhraseSuggestOptions extends SuggestOptions {
```

Comment:
Just picked these two fields to show the difference between phrase and term suggest, can add more or less of the possible fields depending on how strictly typed we want this to be

Reply:
We could also just use a generic type argument here and only have one method, ex:

```typescript
export interface SuggestionBuilder<B> {
  suggest<T = object>(field: string, options: T): B;
  getSuggestions(): object;
}

bb.suggest<PhraseSuggest>('product', { ...optionsForPhrase })
bb.suggest<TermSuggest>('product', { ...optionsForTerms })
```

Reply:
I think the right way to go is `suggestionBuilder().suggest('term', 'products', { text: 'testing' })` over `suggestionBuilder().termSuggest('products', { text: 'testing' })`, since it seems more in line with the remaining bodybuilder API. It's still possible to do strong typing through

```ts
	export interface PhraseSuggestOptions extends SuggestOptions {
		size?: number;
		gram_size?: number;	
	}
	export interface TermSuggestOptions extends SuggestOptions {
		suggest_mode?: "missing" | "popular" | "always"
	}
	export interface SuggestionBuilder<B> {
		suggest<
			SuggestT extends "term" | "phrase",
			SuggestOptionT extends SuggestOptions = SuggestT extends "term" ? TermSuggestOptions : PhraseSuggestOptions
		>(type: SuggestT, field: string, options: SuggestOptionT): B;
		getSuggestions(): object;
	}
```

though this would only enforce that the options _extend_ those types (i.e. you get the type hint e.g. for the `suggest_mode` property, but the value of the property is allowed to be anything, since your literal type will extend the TermSuggestOptions)

Reply:
^ Yeah I thought about this a bit more last night and agree, will swap it out for `suggest` with the generic. I like the multiple type argument approach, so will go that route! I think it's a good middle ground of having much stronger typing in TS, while keeping the JS api untouched